xdsl/dialects/riscv.py

```diff
@@ -331,6 +334,73 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        operands = cls.parse_operands(parser)
+        successors = cls.parse_successors(parser)
+        regions = cls.parse_regions(parser)
+        attributes = cls.parse_attributes(parser)
+        parser.parse_punctuation(":")
+        result_types = cls.parse_function_type(parser).outputs.data
+        return cls.create(
+            operands=operands,
+            result_types=result_types,
+            attributes=attributes,
+            successors=successors,
+            regions=regions,
+        )
+
+    @classmethod
+    def parse_operands(cls, parser: Parser) -> Sequence[SSAValue]:
```

Comment:
All these methods should either be private, or moved to a different place. I would like to be very careful with the public facing API of *all* the riscv operations...

Reply:
I'm not sure if someone write a V-extension, not included in xDSL, they will need to use these APIs.

Reply:
I would not optimize for this case. If they need their own parsing/printing, they can just overwrite the parse/print functions as a whole. Supporting too generic use cases again unnecessarily broadens the scope of our discussion.

Reply:
Removed the function, will be exposed in next PR.